src/js/containers/search/helpers/searchAnalytics.js

```diff
@@ -0,0 +1,235 @@
+            return convertReducibleValue(value, 'Award ID');
+        case 'selectedCFDA':
+            return convertReducibleValue(
+                value,
+                'CFDA Program',
+                (cfda) => `${cfda.program_number} - ${cfda.program_title}`
+            );
+        case 'selectedNAICS':
+            return convertReducibleValue(
+                value,
+                'NAICS Code',
+                (naics) => `${naics.naics} - ${naics.naics_description}`
+            );
+        case 'selectedPSC':
+            return convertReducibleValue(
+                value,
+                'Product/Service Code (PSC)',
+                (psc) => `${psc.product_or_service_code} - ${psc.psc_description}`
+            );
+        case 'pricingType':
+            return convertReducibleValue(
+                value,
+                'Type of Contract Pricing',
+                (pricing) => pricingTypeDefinitions[pricing]
+            );
+        case 'setAside':
+            return convertReducibleValue(
+                value,
+                'Type of Set Aside',
+                (sa) => setAsideDefinitions[sa]
+            );
+        case 'extentCompeted':
+            return convertReducibleValue(
+                value,
+                'Extent Competed',
+                (extent) => extentCompetedDefinitions[extent]
+            );
+        default:
+            return null;
+    }
+};
+
+export const unifyDateFields = (redux) => {
+    // clone the filter reducer so we don't accidentally modify it
+    const clonedRedux = Object.assign({}, redux);
+    // unify the data fields into a single field
+    if (clonedRedux.timePeriodType === 'fy') {
+        clonedRedux.timePeriod = clonedRedux.timePeriodFY;
+    }
+    else {
+        clonedRedux.timePeriod = [clonedRedux.timePeriodStart, clonedRedux.timePeriodEnd];
+    }
+    return clonedRedux;
+};
+
+export const convertFiltersToAnalyticEvents = (redux) => {
+    const filters = unifyDateFields(redux);
+    return Object.keys(filters).reduce((converted, type) => {
+        const value = filters[type];
+        const analyticEvent = convertFilter(type, value);
+        if (analyticEvent) {
+            return converted.concat(analyticEvent);
+        }
+        return converted;
+    }, []);
+};
+
+export const sendAnalyticEvents = (events) => {
+    events.forEach((event) => {
+        Analytics.event(Object.assign({}, event, {
+            category: eventCategory
+        }));
+    });
+};
+
+export const sendFieldCombinations = (events) => {
+    // record the filter field combinations that were selected
+    // extract the action label from each event and then eliminate duplicates
+    const fields = uniq(events.reduce((parsed, event) => {
+        if (event.action) {
+            parsed.push(event.action);
+        }
+        return parsed;
+    }, []));
+
+    Analytics.event({
+        category: 'Advanced Search - Search Fields',
+        action: fields.sort().join('-')
```

Comment:
A different delimiter might make this list easier to read when there are fields with a dash (e.g. `Funding Agency-Time Period - Fiscal Year`).

Reply:
Okay, I can switch to pipes `|`, but I wanted to set up so later on we can programmatically split the strings again.